docs/generate_py_docs.py

```diff
@@ -0,0 +1,174 @@
+import re
+import os
+from textwrap import dedent, indent
+from os import listdir
+from os.path import isfile, join
+from bs4 import BeautifulSoup
+from docutils import core
+
+
+def rst2html(input_string: str, source_path: str = None, destination_path: str = None,
+             input_encoding: str = 'unicode', doctitle: bool = True,
+             initial_header_level: int = 1):
+    overrides = {'input_encoding': input_encoding,
+                 'doctitle_xform': doctitle,
+                 'initial_header_level': initial_header_level}
+    parts = core.publish_parts(
+        source=input_string, source_path=source_path,
+        destination_path=destination_path,
+        writer_name='html', settings_overrides=overrides)
+    return parts['body']
+
+
+def html2txt(html: str):
+    html_elements = BeautifulSoup(html, features="html.parser")
+
+    for tag in html_elements.select('style'):
+        tag.decompose()
+
+    for tag in html_elements.select('td'):
+        # remove non-breaking spaces
+        if tag.text == u'\xa0':
+            tag.decompose()
+        else:
+            # add new line and indent for the cells inside a table
+            if 'class' not in tag.attrs:
+                tag.insert_before('\n      ')
+
+    # substitute double apostrophe with a single apostrophe
```

Comment:
+1 for the comments on what happens in each step

Reply:
Although... here it looks it doesn't describe what happens in this step ;D

Reply:
Hmm, which line are you referring to? @PawelLipski

Reply:
Hmm as the piece of code below this comment:
```python
    # substitute double apostrophe with a single apostrophe
    for tag in html_elements.select('tt'):
        tag.insert_before('`')
        tag.insert_after('`')
```
seems to insert **backticks** around preformatted text (`tt`), nothing to do with apostrophes 🤔

Reply:
@PawelLipski yeah, exactly that, i'll replace `apostrophes` with `backticks` in the comment to make it clearer